src/python/pants/core/util_rules/wrap_source_intergration_test.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from __future__ import annotations
+
+from textwrap import dedent
+
+from pants.testutil.pants_integration_test import run_pants, setup_tmpdir
+
+
+def test_synthesized_python_is_included_in_package() -> None:
+    sources = {
+        "src/BUILD": dedent(
+            """\
+            experimental_shell_command(
+                name="manufacture_python_code",
+                tools=["touch",],
+                command='echo print\\\\(\\\\"Hello, World!\\\\"\\\\) > hello_world.py',
+                execution_dependencies=(),
+                output_files=["hello_world.py",],
+            )
+
+            experimental_wrap_as_python_sources(
+                name="python_dependency",
+                inputs=[":manufacture_python_code"],
+            )
+
+            pex_binary(
+                name="app",
+                dependencies=[":python_dependency"],
+                entry_point=str(build_file_dir()).replace("/", ".") + ".hello_world",
+            )
+            """
+        ),
+    }
+    with setup_tmpdir(sources) as tmpdir:
+        args = [
+            "--backend-packages=['pants.backend.python','pants.backend.shell']",
+            f"--source-root-patterns=['/{tmpdir}/src']",
+            "run",
+            f"{tmpdir}/src:app",
```

Comment:
You really should (also or instead of) test running the source directly.

Reply:
You can't actually run the source directly, which is the same behaviour as codegen in other cases. You can only execute concrete source files.

Reply:
c&v:

```
pants.engine.internals.specs_rules.NoApplicableTargetsException: No applicable files or targets matched. The `run` goal works with these target types:

  * deploy_jar
  * docker_image
  * experimental_run_shell_command
  * go_binary
  * java_source
  * junit_test
  * jvm_artifact
  * pex_binary
  * pyoxidizer_binary
  * python_requirement
  * python_source
  * python_test
  * scala_junit_test
  * scala_source
  * scalatest_test
```

Reply:
Oh thats really unfortunate, since you CAN run a `python_source` target. Seems like a shortcoming of the core code, no?

Reply:
Try without the `PythonRunGoalUseSandboxField` in the required fields of `src/python/pants/backend/python/goals/run_python_source.py`?

Reply:
_probably_?

The counterpoint is that you can't run a `python_sources` target generator, which is a closer analogy to what we're dealing with here. Not the targets generated by that generator, but the generator itself.

I think there's some modelling difficulty in there that would take a bit of work to overcome. Generally speaking, I think codegenned sources are more likely to be library code than runnables, so I'm not certain it's worth the work.

Reply:
> Try without the `PythonRunGoalUseSandboxField` in the required fields of `src/python/pants/backend/python/goals/run_python_source.py`?

I added wrapper rules for Java sources, it should have shown up in that list of target types: it did not.

Reply:
Since you're using `PythonSourceField` I would say it isn't like `pyhton_sources` :wink: 

The reason I suggested trying without `PythonRunGoalUseSandboxField` in the required fields is because I'm wondering if the FieldSet will match this target if it meets all the required fields (just `PythonSourceField`) my gut says so, but that, I think, is something we could change in the core code.

Reply:
Right, I added the rules for Java sources, since the fieldset is already smaller. It doesn't work.

Reply:
Ah right because the target has a field of the (special flavor of) `ActivateWrapSourceTargetField` not of the actual field we're generating :thinking:

Reply:
Right. And that's true of all code generators -- currently the assumption with codegen is that you're producing library code. I think the case I'm demonstrating in the test (running generated code) is marginal, but it also keeps the test small. 

The alternative would be creating a `python_source` and importing the resulting file, however, in that case, Python is perfectly happy to import a python file from a `files` target if you specify the dependency manually. It's only when you get to packaging that the deficiency becomes obvious.

Reply:
Yeah. I think there's certainly room for improvement. But that's isn't this PR :smile: